common/src/main/java/com/wynntils/models/items/encoding/impl/block/DefenseDataTransformer.java

```diff
@@ -0,0 +1,96 @@
+import com.wynntils.models.items.encoding.data.DefenseData;
+import com.wynntils.models.items.encoding.type.DataTransformer;
+import com.wynntils.models.items.encoding.type.ItemTransformingVersion;
+import com.wynntils.utils.UnsignedByteUtils;
+import com.wynntils.utils.type.ArrayReader;
+import com.wynntils.utils.type.ErrorOr;
+import com.wynntils.utils.type.Pair;
+import com.wynntils.utils.type.UnsignedByte;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefenseDataTransformer extends DataTransformer<DefenseData> {
+    @Override
+    protected ErrorOr<UnsignedByte[]> encodeData(ItemTransformingVersion version, DefenseData data) {
+        return switch (version) {
+            case VERSION_1 -> encodeDefenseData(data);
+        };
+    }
+
+    @Override
+    public ErrorOr<DefenseData> decodeData(ItemTransformingVersion version, ArrayReader<UnsignedByte> byteReader) {
+        return switch (version) {
+            case VERSION_1 -> decodeDefenseData(byteReader);
+        };
+    }
+
+    @Override
+    public byte getId() {
+        return 11;
+    }
+
+    @Override
+    protected boolean shouldEncodeData(ItemTransformingVersion version, DefenseData data) {
+        return data.health() != 0 || !data.defences().isEmpty();
+    }
+
+    private ErrorOr<UnsignedByte[]> encodeDefenseData(DefenseData data) {
+        List<UnsignedByte> bytes = new ArrayList<>();
+
+        // The first byte is the length of the integer that the health can fit in.
+        // The next bytes are the health bytes, which are assembled into an integer.
+        UnsignedByte[] unsignedBytes = UnsignedByteUtils.encodeVariableSizedInteger(data.health());
+        bytes.add(UnsignedByte.of((byte) unsignedBytes.length));
+        bytes.addAll(List.of(unsignedBytes));
+
+        // The next byte is the number of defense stats present on the item.
+        bytes.add(UnsignedByte.of((byte) data.defences().size()));
+
+        // A defense stat is encoded the following way:
+        for (Pair<Element, Integer> defence : data.defences()) {
+            // The first byte is the id of the skill (`ETFWA`).
+            bytes.add(UnsignedByte.of((byte) defence.a().ordinal()));
+
+            // The next byte is the length of the integer that the defense can fit in.
+            // The next bytes are the defense bytes, which are assembled into an integer.
+            unsignedBytes = UnsignedByteUtils.encodeVariableSizedInteger(defence.b());
+            bytes.add(UnsignedByte.of((byte) unsignedBytes.length));
+            bytes.addAll(List.of(unsignedBytes));
+        }
+
+        return ErrorOr.of(bytes.toArray(new UnsignedByte[0]));
+    }
+
+    private ErrorOr<DefenseData> decodeDefenseData(ArrayReader<UnsignedByte> byteReader) {
+        // The first byte is the length of the integer that the health can fit in.
+        // The next bytes are the health bytes, which are assembled into an integer.
+        int length = byteReader.read().value();
```

Comment:
This is a common pattern, reading a length byte + a variable sized integer. I recommend introducing a helper method to do just this. Then, if you decide to switch to another method of implementing variable-sized integers, it's just a single method to update.